### PR TITLE
Adding kernbench test

### PR DESCRIPTION
**Pull request by harish-24**
Kernbench is a standard CPU benchmark. Runs a kernel compile and measures the performance by using time command. By default it runs only once with 2 * number of CPUs.

Signed-off-by: Harish <harish@linux.vnet.ibm.com>
Author: Sudeesh John <sudeesh@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by harish-24** [comment]
Last version #35 from Sudeesh

**Comment by harish-24** [comment]
Job log: [kernbench.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/1375251/kernbench.txt)

File: kernel/kernbench.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+from avocado import Test
+from avocado import main
+from avocado.utils import build
+from avocado.utils import process
+from avocado.utils import kernel
+from avocado.utils import cpu
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+log = logging.getLogger('avocado.test')
```
Comment (PraveenPenguin) [inline kernel/kernbench.py]: i think we can drop it as any way it's up to you

File: kernel/kernbench.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            build.make(kern.linux_dir, extra_args='defconfig')
+        else:
+            build.make(kern.linux_dir, extra_args='oldnoconfig')
+        if make_opts:
+            build_string = "/usr/bin/time -o %s make %s -j %s vmlinux" % (
+                timefile, make_opts, threads)
+        else:
+            build_string = "/usr/bin/time -o %s make -j %s vmlinux" % (
+                timefile, threads)
+        process.system(build_string, allow_output_check='None', shell=True)
```
Comment (PraveenPenguin) [inline kernel/kernbench.py]: i think you are interested in time taken on execution (kernel source make ) so using build.run_make is more appropriate library to use here

File: kernel/kernbench.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                         'libcap-ng0', 'libcap-ng-dev', 'elfutils', 'libelf1',
+                         'libnuma-dev', 'libfuse-dev'])
+        elif 'SuSE' in detected_distro.name:
+            deps.extend(['libpopt0', 'glibc', 'glibc-devel',
+                         'popt-devel', 'libcap1', 'libcap-devel',
+                         'libcap-ng-devel'])
+        # FIXME: "redhat" as the distro name for RHEL is deprecated
+        # on Avocado versions >= 50.0.  This is a temporary compatibility
+        # enabler for older runners, but should be removed soon
+        elif detected_distro.name in ['centos', 'fedora', 'rhel', 'redhat']:
```
Comment (PraveenPenguin) [inline kernel/kernbench.py]: i think we can  drop redhat as we are using avocado versions >= 50.0 what you say ?

File: kernel/kernbench.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+            deps.extend(['popt', 'glibc', 'glibc-devel', 'libcap-ng',
+                         'libcap', 'libcap-devel', 'elfutils-libelf',
+                         'elfutils-libelf-devel', 'openssl-devel'])
+
+        for package in deps:
+            if not smg.check_installed(package) and not smg.install(package):
+                self.cancel(package + ' is needed for the test to be run')
```
Comment (PraveenPenguin) [inline kernel/kernbench.py]: using `+` to containing string not looks good can it be like 

self.cancel(' %s is needed for the test to be run' %package )

File: kernel/kernbench.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+            build.make(kern.linux_dir, extra_args='oldnoconfig')
+        if make_opts:
+            build_string = "/usr/bin/time -o %s make %s -j %s vmlinux" % (
+                timefile, make_opts, threads)
+        else:
+            build_string = "/usr/bin/time -o %s make -j %s vmlinux" % (
+                timefile, threads)
+        process.system(build_string, allow_output_check='None', shell=True)
+        if not os.path.isfile('vmlinux'):
+            self.error("No vmlinux found, kernel build failed")
```
Comment (PraveenPenguin) [inline kernel/kernbench.py]: can it be self.fail ?

File: kernel/kernbench.py [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    def test(self):
+        """
+        Kernel build Test
+        """
+        # Initialize kernel object
+        kern = kernel.KernelBuild(
+            self.version, config_path=self.config_path,
+            work_dir=None, data_dirs=None)
+        # Setting the default kernel URL
+        kern.URL = 'https://www.kernel.org/pub/linux/kernel/v4.x/'
```
Comment (PraveenPenguin) [inline kernel/kernbench.py]: i think all from line no 128 to 134 can be wrap in such way as using fetch_asset , than extract in your kernel build path 

```
location = ["https://github.com/torvalds/linux/archive/master.zip"]
        tarball = self.fetch_asset("kselftest.zip", locations=location,
                                   expire='1d')
        archive.extract(tarball, self.srcdir)
        self.buldir = os.path.join(self.srcdir, 'linux-master')
        self.sourcedir = os.path.join(self.buldir, self.testdir)
```

as well url can be managed from yaml file

File: kernel/kernbench.py (hunk 2 above)
Comment (harish-24) [reply kernel/kernbench.py]: `run_make` enough for only the running time of command, but `time` also calculates user, system and elapsed time.

File: kernel/kernbench.py (hunk 1 above)
Comment (harish-24) [reply kernel/kernbench.py]: sure

File: kernel/kernbench.py (hunk 5 above)
Comment (harish-24) [reply kernel/kernbench.py]: sure

File: kernel/kernbench.py (hunk 3 above)
Comment (harish-24) [reply kernel/kernbench.py]: missed that. will remove `redhat`.

File: kernel/kernbench.py (hunk 4 above)
Comment (harish-24) [reply kernel/kernbench.py]: sure

File: kernel/kernbench.py (hunk 2 above)
Comment (PraveenPenguin) [reply kernel/kernbench.py]: sure

File: kernel/kernbench.py (hunk 6 above)
Comment (harish-24) [reply kernel/kernbench.py]: sure sir

**Comment by harish-24** [comment]
@PraveenPenguin Closing this. Opened v2: #660

**Pull request closed without merging**